catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -1323,6 +1342,24 @@ private CswRecordCollection queryProductById(String id) throws CswException {
         return cswRecordCollection;
     }
 
+    protected long getRange(String rangeValue) throws UnsupportedQueryException {
+        long response = -1;
+
+        if (StringUtils.isNotBlank(rangeValue)) {
+            if (rangeValue.startsWith(CswConstants.BYTES_EQUAL)) {
+                String tempString = rangeValue.substring(CswConstants.BYTES_EQUAL.length());
+                List<String> range = Splitter.on('-')
+                        .splitToList(tempString);
+                if (!range.isEmpty() && range.size() <= 2) {
+                    return Long.parseLong(range.get(0));
```

Comment:
I think you should check that `range.get(0)` is a numeric before trying to parse it. If it is not, then you can throw your checked `UnsupportedQueryException`; otherwise, you'll be throwing a runtime exception on this line.